src/lib.rs

```diff
@@ -124,60 +199,100 @@ impl CachedQueries {
                 !attempted_parsings.is_empty(),
                 "Should've tried to parse in at least one language or else should've already failed on no candidate files"
             );
-            if attempted_parsings.len() == 1 {
-                let (supported_language, once_lock) = &attempted_parsings[0];
-                fail(&format!(
-                    "couldn't parse query for {:?}: {}",
-                    supported_language,
-                    once_lock.get().unwrap().as_ref().unwrap_err()
-                ));
-            } else {
-                let mut attempted_parsings = attempted_parsings
+            return Err(Error::NoSuccessfulQueryParsing(
+                attempted_parsings
                     .into_iter()
-                    .map(|(supported_language, _)| format!("{supported_language:?}"))
-                    .collect::<Vec<_>>();
-                attempted_parsings.sort();
-                fail(&format!(
-                    "couldn't parse query for {}",
-                    join_with_or(&attempted_parsings)
-                ));
-            }
+                    .map(|(supported_language, once_lock)| {
+                        (
+                            supported_language,
+                            format!("{}", once_lock.get().unwrap().as_ref().unwrap_err()),
```

Comment:
Maybe you could implement `From` it to your error type to avoid having this code live so far from the error type?

Reply:
Ok I did make this consume `self` so that I could consume the inner `QueryError`'s and store them

Are you saying implement `From<CachedQueries> for Error` and have that contain the logic for doing the consuming-`self`-into-a-`Vec`-of-`SupportedLanguage`'s-and-`QueryError`'s?

If so I could see that but my instinct might be that using `From` to express that is actually a little bit "muddying" since conceptually there might be more than one reason a `CachedQueries` could initialize an `Error` and having it "inline" here it's clearer when/why we're doing that type-mapping?